be/src/exec/vectorized/schema_scan_node.cpp

```diff
@@ -283,4 +288,19 @@ Status SchemaScanNode::set_scan_ranges(const std::vector<TScanRangeParams>& scan
     return Status::OK();
 }
 
+std::vector<std::shared_ptr<pipeline::OperatorFactory>> SchemaScanNode::decompose_to_pipeline(
+        pipeline::PipelineBuilderContext* context) {
+    // the dop of SchemaScanOperator should always be 1.
+    size_t dop = 1;
+
+    size_t buffer_capacity = pipeline::ScanOperator::max_buffer_capacity() * dop;
+    int64_t mem_limit = runtime_state()->query_mem_tracker_ptr()->limit() * config::scan_use_query_mem_ratio;
+    pipeline::ChunkBufferLimiterPtr buffer_limiter = std::make_unique<pipeline::DynamicChunkBufferLimiter>(
+            buffer_capacity, buffer_capacity, mem_limit, runtime_state()->chunk_size());
```

Comment:
is there any check of validation for `config::scan_use_query_mem_ratio`?

Reply:
> is there any check of validation for `config::scan_use_query_mem_ratio`?

I'm afraid there isn't. I think the verification of configurations should be handled centrally when the BE process starts. we can improve it later